src/main/java/core/mate/academy/model/Bulldozer.java

```diff
@@ -1,15 +1,33 @@
 package core.mate.academy.model;
 
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Add some custom fields that could be only in Bulldozer
  * Do not remove no-args constructor
  */
-public class Bulldozer extends Machine {
+public class Bulldozer extends Machine implements MachineProducer<Bulldozer> {
+    private int enginePower;
+
     public Bulldozer() {
     }
 
+    public Bulldozer(int enginePower) {
+        this.enginePower = enginePower;
+    }
+
     @Override
     public void doWork() {
         System.out.println("Bulldozer started to work");
     }
+
+    @Override
+    public List<Bulldozer> get() {
+        List<Bulldozer> bulldozers = new ArrayList<>();
+        bulldozers.add(new Bulldozer(90));
+        bulldozers.add(new Bulldozer(100));
+        return bulldozers;
+    }
```

Comment:
like previous mentor already said, you dont have to implement Producer logic onto model classes, instead of this create `BulldozerProducer`, `ExcavatorProducer`... classes in `service` package and implement according interfaces onto them
```suggestion
public class Bulldozer extends Machine {
    private int enginePower;

    public Bulldozer() {
    }

    public Bulldozer(int enginePower) {
        this.enginePower = enginePower;
    }

    @Override
    public void doWork() {
        System.out.println("Bulldozer started to work");
    }
```